hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/ReactiveImprovedExtractionContextImpl.java

```diff
@@ -65,54 +62,42 @@ public ReactiveImprovedExtractionContextImpl(
 				NoopDdlTransactionIsolator.INSTANCE,
 				databaseObjectAccess
 		);
-		service = registry.getService( ReactiveConnectionPool.class );
+		connectionPool = registry.getService( ReactiveConnectionPool.class );
 	}
 
 	@Override
 	public <T> T getQueryResults(
 			String queryString,
 			Object[] positionalParameters,
 			ResultSetProcessor<T> resultSetProcessor) throws SQLException {
-
-		final CompletionStage<ReactiveConnection> connectionStage = service.getConnection();
-
-		try (final ResultSet resultSet = getQueryResultSet( queryString, positionalParameters, connectionStage )) {
+		try (final ResultSet resultSet = getQueryResultSet( queryString, positionalParameters )) {
 			return resultSetProcessor.process( resultSet );
 		}
-		finally {
-			// This method doesn't return a reactive type, so we start closing the connection and ignore the result
-			connectionStage
-					.handle( ReactiveImprovedExtractionContextImpl::ignoreException )
-					.thenCompose( ReactiveImprovedExtractionContextImpl::closeConnection );
-
-		}
-	}
-
-	private static ReactiveConnection ignoreException(ReactiveConnection reactiveConnection, Throwable throwable) {
-		return reactiveConnection;
-	}
-
-	private static CompletionStage<Void> closeConnection(ReactiveConnection connection) {
-		// Avoid NullPointerException if we couldn't create a connection
-		return connection != null ? connection.close() : voidFuture();
 	}
 
 	private ResultSet getQueryResultSet(
 			String queryString,
-			Object[] positionalParameters,
-			CompletionStage<ReactiveConnection> connectionStage) {
+			Object[] positionalParameters) {
 		final Object[] parametersToUse = positionalParameters != null ? positionalParameters : new Object[0];
-		final Parameters parametersDialectSpecific = Parameters.instance(
-				getJdbcEnvironment().getDialect()
-		);
+		final Parameters parametersDialectSpecific = Parameters.instance( getJdbcEnvironment().getDialect() );
 		final String queryToUse = parametersDialectSpecific.process( queryString, parametersToUse.length );
-		return connectionStage.thenCompose( c -> c.selectJdbcOutsideTransaction( queryToUse, parametersToUse ) )
+		return connectionPool
+				// DDL needs to run outside the current transaction. For example:
+				// - increment on a table-based id generator should happen outside the current tx.
+				// - not all databases support transactional DDL
+				.selectJdbcOutsideTransaction( queryToUse, parametersToUse )
```

Comment:
This makes sense, but then:

1. I don't think this code is ever invoked within a transaction... is it?
2. Why wouldn't this code just retrieve a connection from the pool and use it directly? I.e. instead of removing the retrieval of a connection (`service.getConnection()`), you'd remove the call to `selectJdbcOutsideTransaction`, and would just use the connection to perform DDL. I think this would achieve the same results, without needing a specific method in the pool... ?

Reply:
I've answered in the other comments bu, shortly:
1.  I like that it's more explicit about what's going on for future reference (I have the memory of a gold fish)
2. I think the code is simpler this way, but I'm open about changing it. See https://github.com/hibernate/hibernate-reactive/pull/1928#discussion_r1624264056